core/server/src/main/java/alluxio/worker/block/evictor/LIRSEvictor.java

```diff
@@ -0,0 +1,573 @@
+    SpaceContainer spaceContainer = mSpaceManager.get(location);
+    Iterator<Map.Entry<Pair<BlockStoreLocation, Long>, BlockLIRSInfo>> it =
+        mLIRCache.entrySet().iterator();
+    if (!it.hasNext()) {
+      return;
+    }
+    Entry<Pair<BlockStoreLocation, Long>, BlockLIRSInfo> entry = it.next();
+    BlockLIRSInfo blockinfo = entry.getValue();
+    while (spaceContainer.getLIRBytes() > spaceContainer.getCapacity() * mLIRPercent
+        || !entry.getKey().getFirst().equals(location) || blockinfo.isHIR()
+        || !blockinfo.isResident()) {
+      if (!entry.getKey().getFirst().equals(location)) {
+        if (it.hasNext()) {
+          entry = it.next();
+          blockinfo = entry.getValue();
+          continue;
+        } else {
+          break;
+        }
+      } else if (blockinfo.isHIR() || !blockinfo.isResident()) {
+        it.remove();
+      } else if (blockinfo.isLIR()) {
+        long blockSize = mBlockIdToSize.get(entry.getKey().getSecond());
+        blockinfo.setIsHIR(true);
+        blockinfo.setResident(true);
+        spaceContainer.moveBlockFromLIRToHIR(blockSize);
+        mHIRCache.put(entry.getKey(), blockinfo);
+        it.remove();
+      }
+      if (it.hasNext()) {
+        entry = it.next();
+        blockinfo = entry.getValue();
+      } else {
+        break;
+      }
+    }
+    mSpaceManager.put(location, spaceContainer);
+  }
+
+  /**
+   * Class to record the state of a block. There are 3 kinds of state of a block: HIR block: an HIR
+   * block which will be evicted first; LIR block: a LIR block which will be evicted after HIR
+   * blocks; Ghost block (not resident on the tier): a block which has been moved to other tiers,
+   * but an entry is maintained to mark when it is last accessed.
+   */
+  class BlockLIRSInfo {
+    // If the block is an HIR block
+    private boolean mIsHIR;
+    // If the block is an LIR block
+    private boolean mIsLIR;
+    // If the block is a ghost block (resident on the tier)
+    private boolean mIsResident;
+
+    public BlockLIRSInfo(boolean isLIR, boolean isResident) {
+      mIsLIR = isLIR;
+      mIsHIR = !isLIR;
+      mIsResident = isResident;
+    }
+
+    public boolean isHIR() {
+      return mIsHIR;
+    }
+
+    public boolean isLIR() {
+      return mIsLIR;
+    }
+
+    public boolean isResident() {
+      return mIsResident;
+    }
+
+    public void setResident(boolean isResident) {
+      mIsResident = isResident;
+    }
+
+    public void setIsHIR(boolean isHIR) {
+      mIsLIR = !isHIR;
+      mIsHIR = isHIR;
+    }
+
+    public void setIsLIR(boolean isLIR) {
+      mIsLIR = isLIR;
+      mIsHIR = !isLIR;
+    }
+  }
+
+  /**
+   * Class to record the space used condition of a StorageDir.
+   */
+  class SpaceContainer {
+    // Used space of HIR blocks
+    private long mHIRBytes;
+    // Used space of LIR blocks
+    private long mLIRBytes;
+    // Capacity space of the StorageDir
+    private long mCapacity;

```

Comment:
@uronce-cc This has nothing to do with temp cache. `SpaceContainer` is used to record how much space of HIR and LIR cache have used in order to control the space under limit.